queries/dmpworks/python/dmpworks/batch/tasks.py

```diff
@@ -10,51 +12,60 @@
 
 @dataclass
 class DownloadTaskContext:
-    local_dir: pathlib.Path
+    download_dir: pathlib.Path
     target_uri: str
 
 
 @contextmanager
-def download_source_task(bucket_name: str, dataset: str, task_id: str):
+def download_source_task(bucket_name: str, dataset: str, task_id: str) -> Generator[DownloadTaskContext, Any, None]:
     target_uri = s3_uri(bucket_name, dataset, task_id, "download")
-    local_dir = local_path(dataset, task_id, "download")
+    download_dir = local_path(dataset, task_id, "download")
 
     clean_s3_prefix(target_uri)
 
     log.info(f"Downloading {dataset}")
     ctx = DownloadTaskContext(
-        local_dir=local_dir,
+        download_dir=download_dir,
         target_uri=target_uri,
     )
     yield ctx
 
-    upload_to_s3(local_dir, target_uri)
+    upload_to_s3(download_dir, target_uri)
+
+    # Cleanup files as we can't guarantee that we will end up on the same worker
+    # again, and we don't want to take disk space that other tasks might use
+    shutil.rmtree(download_dir, ignore_errors=True)
 
 
 @dataclass
 class TransformTaskContext:
-    in_dir: pathlib.Path
-    out_dir: pathlib.Path
+    download_dir: pathlib.Path
+    transform_dir: pathlib.Path
     target_uri: str
 
 
 @contextmanager
-def transform_parquets_task(bucket_name: str, dataset: str, task_id: str):
-    in_dir = local_path(dataset, task_id, "transform")
-    out_dir = local_path(dataset, task_id, "transform")
+def transform_parquets_task(bucket_name: str, dataset: str, task_id: str) -> Generator[TransformTaskContext, Any, None]:
+    download_dir = local_path(dataset, task_id, "download")
+    transform_dir = local_path(dataset, task_id, "transform")
     target_uri = s3_uri(bucket_name, dataset, task_id, "transform")
 
     clean_s3_prefix(target_uri)
 
-    download_from_s3(f"{s3_uri(bucket_name, dataset, task_id, "download")}*", in_dir)
-    out_dir.mkdir(parents=True, exist_ok=True)
+    download_from_s3(f"{s3_uri(bucket_name, dataset, task_id, "download")}*", download_dir)
+    transform_dir.mkdir(parents=True, exist_ok=True)
 
     log.info(f"Transforming {dataset}")
     ctx = TransformTaskContext(
-        in_dir=in_dir,
-        out_dir=out_dir,
+        download_dir=download_dir,
+        transform_dir=transform_dir,
         target_uri=target_uri,
     )
     yield ctx
 
-    upload_to_s3(out_dir / "parquets", f"{target_uri}parquets/", "*.parquet")
+    upload_to_s3(transform_dir / "parquets", f"{target_uri}parquets/", "*.parquet")
```

Comment:
Cool, I didn't realize that Pathlib had the syntactic sugar with the slash since I think I used some other method that was more cumbersome in the past.

Reply:
Yeah it is quite handy, I had always used `os.path.join` before and decided I should try the newer way after seeing someone else use it.